src/topic.rs

```diff
@@ -0,0 +1,67 @@
+use std::fmt::{Debug, Display, Formatter};
+
+use flexstr::{SharedStr, ToSharedStr};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+#[repr(transparent)]
+pub struct Topic(SharedStr);
+
+impl Debug for Topic {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        if f.alternate() {
+            write!(f, "{:#?}", self.0)
+        } else {
+            write!(f, "Topic({:#?})", self.0)
+        }
+    }
+}
+impl Display for Topic {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+impl From<&str> for Topic {
```

Comment:
`From<&'str>`, `From<String>` and `FromSharedStr>` can be consolidated with `impl T From<T> for Topic where T: Into<SharedStr>`